skimage/morphology/__init__.pyi

```diff
@@ -0,0 +1,78 @@
+# Explicitly setting `__all__` is necessary for type inference engines
+# to know which symbols are exported. See
+# https://peps.python.org/pep-0484/#stub-files
+
+__all__ = [
+    'area_closing',
+    'area_opening',
+    'ball',
+    'binary_closing',
+    'binary_dilation',
+    'binary_erosion',
+    'binary_opening',
+    'black_tophat',
+    'closing',
+    'convex_hull_image',
+    'convex_hull_object',
+    'cube',
+    'diameter_closing',
+    'diameter_opening',
+    'diamond',
+    'dilation',
+    'disk',
+    'ellipse',
+    'erosion',
+    'flood',
+    'flood_fill',
+    'footprint_from_sequence',
+    'h_maxima',
+    'h_minima',
+    'isotropic_closing',
+    'isotropic_dilation',
+    'isotropic_erosion',
+    'isotropic_opening',
+    'label',
+    'local_maxima',
+    'local_minima',
+    'max_tree',
+    'max_tree_local_maxima',
+    'medial_axis',
+    'octagon',
+    'octahedron',
+    'opening',
+    'reconstruction',
+    'rectangle',
+    'remove_small_holes',
+    'remove_small_objects',
+    'skeletonize',
+    'skeletonize_3d',
+    'square',
+    'star',
+    'thin',
+    'white_tophat'
+]
+
+from .binary import (binary_closing, binary_dilation, binary_erosion,
+                     binary_opening)
+from .gray import (black_tophat, closing, dilation, erosion, opening,
+                   white_tophat)
+from .isotropic import (isotropic_erosion, isotropic_dilation,
+                        isotropic_opening, isotropic_closing)
+from .footprints import (ball, cube, diamond, disk, ellipse,
+                         footprint_from_sequence, octagon, octahedron,
+                         rectangle, square, star)
+from ..measure._label import label
```

Comment:
I wonder if this `..` is tripping things up. I don't see anything else strange in the file.

Reply:
Thanks @stefanv 
I wonder too

Reply:
No, I think it's https://github.com/scikit-image/scikit-image/blob/a7e7f46025006c65418e70ba063b6ddabef35b35/skimage/morphology/__init__.pyi#L64

which is tripping up `lazy_loader` and hitting [`lazy_loader/__init__.py#L204-L206`](https://github.com/scientific-python/lazy_loader/blob/2334bd279d40e0dadd3af48fe4ec494d3ce7f47d/lazy_loader/__init__.py#L204-L206). I can reproduce that locally. Changing to `.measure._label import label doesn't work. But if I comment out the line the imports and tests run as expected.

Reply:
A somewhat unsatisfactory workaround would be to move that line to a submodule in `skimage.morphology`. E.g. apply this patch

```diff
Subject: [PATCH] Avoid within-module import
---
Index: skimage/morphology/__init__.pyi
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/skimage/morphology/__init__.pyi b/skimage/morphology/__init__.pyi
--- a/skimage/morphology/__init__.pyi	(revision a7e7f46025006c65418e70ba063b6ddabef35b35)
+++ b/skimage/morphology/__init__.pyi	(date 1686130255245)
@@ -61,7 +61,6 @@
 from .footprints import (ball, cube, diamond, disk, ellipse,
                          footprint_from_sequence, octagon, octahedron,
                          rectangle, square, star)
-from skimage.measure._label import label
 from ._skeletonize import medial_axis, skeletonize, skeletonize_3d, thin
 from .convex_hull import convex_hull_image, convex_hull_object
 from .grayreconstruct import reconstruction
@@ -76,3 +75,4 @@
     max_tree,
     max_tree_local_maxima
 )
+from ._util import label
Index: skimage/morphology/_util.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/skimage/morphology/_util.py b/skimage/morphology/_util.py
--- a/skimage/morphology/_util.py	(revision a7e7f46025006c65418e70ba063b6ddabef35b35)
+++ b/skimage/morphology/_util.py	(date 1686130377270)
@@ -4,6 +4,8 @@
 import numpy as np
 from scipy import ndimage as ndi
 
+from skimage.measure import label
+
 
 def _validate_connectivity(image_dim, connectivity, offset):
     """Convert any valid connectivity to a footprint and offset.

```
---

Do we want to keep `label` available in both the `measure` and `morphology` module? I feel like that goes against the "there should be one obvious way" ideal just for the sake of some convenience? I would probably remove that function from `__all__` in `skimage.morphology` but keep it around maybe with an added deprecation? But that should be another PR.

Reply:
Thanks @lagru 
I will remove the `skimage.measure._label import label` for the `pyi` file.

Reply:
I may misunderstand you, but that is not what I was encouraging with https://github.com/scikit-image/scikit-image/pull/7007#discussion_r1221277104. Rather you need to replace the line with `from ._util import label` and modify `skimage.morphology._util` accordingly (have a look at the patch).

Reply:
Sorry for the misunderstanding. I updated my comment while you were replying to it. :)
I will apply your suggestions accordingly.

Reply:
No, problem at all. I just got confused by your well timed edits. :sweat_smile: